sherpa/fit.py

```diff
@@ -470,9 +471,36 @@ def __init__(self, data, model, stat, method, itermethod_opts={'name': 'none'}):
     def _sig_handler(self, signum, frame):
         raise KeyboardInterrupt()
 
+    def get_bkg_data(self, dep):
+        """get the bkg data for wstat and the user defined statistics"""
+
+        bkg_dep = []
+        data_size = None
+        exposure_time = None
+
+        len_datasets = len(self.data.datasets)
+
+        data_size = zeros(len_datasets, dtype=int)
+        exposure_time = zeros(2 * len_datasets)
+        for index in xrange(len_datasets):
+            mydata = self.data.datasets[index]
+            if hasattr(mydata, 'response_ids') and \
+                    hasattr(mydata, 'background_ids') and \
+                    len(mydata.response_ids) and len(mydata.background_ids):
+                bkg = mydata.get_background(mydata.background_ids[0])
+                tmp_bkg_dep = bkg.get_dep(True)
+                data_size[index] = tmp_bkg_dep.size
+                bkg_dep = append(bkg_dep, tmp_bkg_dep)
+                exposure_time[2 * index] = mydata.exposure
+                exposure_time[2 * index + 1] = bkg.exposure
```

Comment:
For the wstat case, a simple hack to include the BACKSCAL ratios, would be

```
                try:
                    bexp = bkg.exposure * bkg.backscal / mydata.backscal
                except AttributeError:
                    bexp = bkg.exposure
                exposure_time[2 * index + 1] = bexp
```

which seems to give me the same stat values as XSPEC does. I do not like this solution, as (as also mentioned by @cdell) this feels like trying to work around the limitations of an interface, but I don't have any concrete suggestions for improvements.